packit_service/service/events.py

```diff
@@ -635,3 +641,119 @@ def get_copr_build_logs_url(event: CoprBuildEvent) -> str:
         f"{event.project_name}/{event.chroot}/"
         f"{event.build_id:08d}-{event.pkg}/builder-live.log.gz"
     )
+
+
+class AbstractPagureEvent(Event, PagurePackageConfigGetter):
+    def __init__(self, trigger: TheJobTriggerType, project_url: str):
+        super().__init__(trigger)
+        self.project_url: str = project_url
+        self.git_ref: Optional[str] = None  # git ref that can be 'git checkout'-ed
+        self.identifier: Optional[str] = (
+            None  # will be shown to users -- e.g. in logs or in the copr-project name
+        )
+
+    def get_project(self) -> GitProject:
+        return ServiceConfig.get_service_config().get_project(url=self.project_url)
+
+
+class PushPagureEvent(AbstractPagureEvent):
```

Comment:
Those event classes look very similar to Github ones:
- It would be nice to merge them.
- We can make a subclass just for the differentiation without any code change if we really want to differentiate that.
  -  Since we have the `https_url`, we can still be able to get service easily.

Reply:
There is different ConfigGetter class implementation and this should be object of refactoring - this shouldn't be part of Event (its breaking SRP and creating tight coupling). This needs discussion and also i think it is out of scope of this PR.